### DISCUSSION TITLE
Search with Opencodelist builder only returns inactive SNOMED codes when there should also be active codes

### DISCUSSION THREAD

**Discussion by doro-s** [opensafely/documentation#755] [Codelists] [answered]
I am trying to build a SNOMED codelist for overweight and obesity in opencodelists.org (see my draft [here](https://www.opencodelists.org/builder/71ebb061/)). NB I first searched for existing obesity lists on opencodelists, but they only include findings, I would like to also include disorder codes.
To build my own codelist, I searched for the terms: obesity, obese, overweight, BMI, Body Mass Index. My problem is that all the key disorder codes indicating obesity that the opencodelist builder returns are inactive ones. The only codes the search returns in the [active] “disorder” section are conditions where obesity is only mentioned as an additional feature (e.g., Diabetes type 2 in obese). For codes indicating obesity itself such as “Obesity” or “Buffalo obesity”, the search only returns inactive codes (e.g., Obesity 154776002, Obesity 190963004, Buffalo obesity 123291007 ) but it doesn’t return any of the active codes for the same term that I find on the NHS [SNOMED CT browser](https://termbrowser.nhs.uk/?perspective=full&conceptId1=190963004&edition=uk-edition&release=v20220511&server=https://termbrowser.nhs.uk/sct-browser-api/snomed&langRefset=999001261000000100,999000691000001104), e.g. 414916001 Obesity (disorder) or Buffalo obesity (disorder) 360566006.
Can someone please tell me where I am going wrong?

**Comment by HelenCEBM** [comment, marked as answer]
Hi, 

I can't fully explain but it looks like when searching for the term "obesity", the code `414916001` Obesity (disorder) does come up, under Disorders...
![image](https://user-images.githubusercontent.com/24392156/170655700-87be6173-3bec-44ba-8974-e1c4dbdc97a8.png)

...however after adding the search for "obese" this code gets re-grouped under Obese in Findings
![image](https://user-images.githubusercontent.com/24392156/170656185-e29d6b48-191a-40dd-b23c-7fd272449ee5.png)

I presume this is related to the complex structure of SNOMED code trees!

However, I am not sure why Buffalo obesity (disorder) `360566006` doesn't seem to appear when searching for "obesity". If you search for the code, or the term "buffalo obesity" it does come up. 

Perhaps @inglesp can help here.

**Reply by doro-s** [reply]
Thank you so much, Helen, I hadn't seen this! Now that you have pointed me to obesity (414916001) I also find all the other terms. Buffalo obesity is grouped under "Obesity by contributing factors". 
I can see now that the probably reason for all the inactive obesity codes is precisely that the position of these codes in the SNOMED structure changed: Both the inactive obesity and Buffalo obesity code do not have any parents (and hence show in opencodelists under "[inactive] disorder"), while the new codes have "Obese (finding)", or as parent (or indirect parent) and therefore opencodelists doesn't show them under disorders.
